src/python/pants/backend/python/subsystems/python_setup.py

```diff
@@ -75,6 +78,8 @@ def register_options(cls, register):
               'NOTE: this keyword is a temporary fix and will be reverted per: '
               'https://github.com/pantsbuild/pants/issues/5696. The long term '
               'solution is tracked by: https://github.com/pantsbuild/pex/issues/456.')
+    register('--build', advanced=True, type=bool, default=UnsetBool, fingerprint= True,
```

Comment:
in pants' context, `build` is a pretty overloaded term - so it might be good to make this more specific both here and at the target level by calling this e.g. `--allow-sdist-builds`?